File: momose/src/parsers/ConfigSimParser.java
package parsers;

import java.io.File;
import java.util.Vector;

import org.xml.sax.Attributes;

import engine.Scenario;
import engine.SimTime;

public class ConfigSimParser extends TimeScenarioParser
 {
  private Scenario scenario;
  private SimTime simTime;
  private Vector modBuildNames;
  private Vector recBuildNames;
  
  private boolean configFile;
  
  public ConfigSimParser(File file)
   {
	scenario=null;
	simTime=null;
	modBuildNames=new Vector();
	recBuildNames=new Vector();
	configFile=false;
   }//Fine costruttore
   
  public Vector getModBuildNames()
   {return modBuildNames;}
  
  public Vector getRecBuildNames()
   {return recBuildNames;}
  
  public boolean isConfigFile()
   {return configFile;}
  
  public void startElement (String uri, String name,String qName, Attributes atts)
   {
	//Richiamo il metodo del genitore per gestire il tempo e lo scenario  
    super.startElement(uri, name, qName, atts);  
	
    //E' un file di configurazione??
    if(qName.equals("ConfigInfo"))
	 { configFile=true; }  
    
	//Nome di un model builder
	if(qName.equals("ModBuilder"))
	 { addModBuildName(atts); }  
	
	//Nome du un recorderBuilder
	if(qName.equals("RecBuilder"))
	 { addRecBuildName(atts); }  
	
  }//Fine startElement
  
 private void addModBuildName(Attributes atts) 
  {
   String name=atts.getValue(0);
   if(name!=null)
	 modBuildNames.add(name);  
  }//Fine addModBuildName 

 private void addRecBuildName(Attributes atts) 
  {
   String name=atts.getValue(0);
   if(name!=null)
	 recBuildNames.add(name);  
  }//Fine addRecBuildName
}//Fine ConfigSimParser

File: momosecpp/src/ConfigSimParser.cpp
#include "ConfigSimParser.h"

#include <string>

ConfigSimParser::ConfigSimParser():TimeScenarioParser()
 {
   configFile=false;	 
 }//Fine costruttore

ConfigSimParser::~ConfigSimParser(){}


void ConfigSimParser::OnStartElement(const XML_Char* name,const XML_Char** attrs)
  {
    //Richiamo il metodo del genitore per gestire il tempo e lo scenario  
    TimeScenarioParser::OnStartElement(name,attrs);	
	
     string str=xmlCharVectToString(name);	     
	  
    //E' un file di configurazione??
    if(str=="ConfigInfo")
	 { configFile=true; }  
    
	//Nome di un model builder
    if(str=="ModBuilder")
      { addModBuildName(attrs); }  
	
     //Nome du un recorderBuilder
     if(str=="RecBuilder")
	{ addRecBuildName(attrs); }  
  }//Fine OnStartElement
  
  
void ConfigSimParser::addModBuildName(const XML_Char** attrs) 
  {
     string name=getAttrFromName(attrs,"name");  
     if(name!="")
	 modBuildNames.push_back(name);  
  }//Fine addModBuildName 

void ConfigSimParser::addRecBuildName(const XML_Char** attrs) 
  {
    string name=getAttrFromName(attrs,"name");
    if(name!="")
	 recBuildNames.push_back(name);  
  }//Fine addRecBuildName  
 
 
void ConfigSimParser::OnCharacterData(const XML_Char* data, int len){}
void ConfigSimParser::OnEndElement(const XML_Char* name){}